tests/evaluate/logging/utils.py

```diff
@@ -0,0 +1,230 @@
+from typing import List
+
+import pandas as pd
+
+from mlflow.entities.evaluation import Evaluation as EvaluationEntity
+from mlflow.evaluation.utils import (
+    _get_assessments_dataframe_schema,
+    _get_evaluations_dataframe_schema,
+    _get_metrics_dataframe_schema,
+    _get_tags_dataframe_schema,
+)
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INTERNAL_ERROR, RESOURCE_DOES_NOT_EXIST
+from mlflow.tracking.client import MlflowClient
+
+
+def get_evaluation(*, run_id: str, evaluation_id: str) -> EvaluationEntity:
+    """
+    Retrieves an Evaluation object from an MLflow Run.
+
+    Args:
+        run_id (str): ID of the MLflow Run containing the evaluation.
+        evaluation_id (str): The ID of the evaluation.
+
+    Returns:
+        Evaluation: The Evaluation object.
+    """
+    client = MlflowClient()
+    if not _contains_evaluation_artifacts(client=client, run_id=run_id):
+        raise MlflowException(
+            "The specified run does not contain any evaluations. "
+            "Please log evaluations to the run before retrieving them.",
+            error_code=RESOURCE_DOES_NOT_EXIST,
+        )
+
+    evaluations_file = client.download_artifacts(run_id=run_id, path="_evaluations.json")
+    evaluations_df = _read_evaluations_dataframe(evaluations_file)
+
+    assessments_file = client.download_artifacts(run_id=run_id, path="_assessments.json")
+    assessments_df = _read_assessments_dataframe(assessments_file)
+
+    metrics_file = client.download_artifacts(run_id=run_id, path="_metrics.json")
+    metrics_df = _read_metrics_dataframe(metrics_file)
+
+    tags_file = client.download_artifacts(run_id=run_id, path="_tags.json")
+    tags_df = _read_tags_dataframe(tags_file)
+
+    return _get_evaluation_from_dataframes(
+        run_id=run_id,
+        evaluation_id=evaluation_id,
+        evaluations_df=evaluations_df,
+        metrics_df=metrics_df,
+        assessments_df=assessments_df,
+        tags_df=tags_df,
+    )
+
+
+def _contains_evaluation_artifacts(*, client: MlflowClient, run_id: str) -> bool:
+    return (
+        any(file.path == "_evaluations.json" for file in client.list_artifacts(run_id))
+        and any(file.path == "_metrics.json" for file in client.list_artifacts(run_id))
+        and any(file.path == "_assessments.json" for file in client.list_artifacts(run_id))
+    )
```

Comment:
/ai rewrite this using `set`

Reply:
/ai cool, make it one-liner

Reply:
Thanks for the suggestion! I've applied it :)